lib/pos_blockchain.go

```diff
@@ -76,17 +76,33 @@ func (bc *Blockchain) processBlockPoS(block *MsgDeSoBlock, currentView uint64, v
 		return false, false, nil, errors.Wrap(err, "processBlockPoS: Problem applying new tip: ")
 	}
 
-	// If the incoming block is not applied as the new tip, we return early.
-	if !appliedNewTip {
-		return false, false, nil, nil
+	// 5. Commit grandparent if possible. Only need to do this if we applied a new tip.
+	if appliedNewTip {
+		if err = bc.runCommitRuleOnBestChain(); err != nil {
+			return false, false, nil, errors.Wrap(err, "processBlockPoS: error committing grandparents: ")
+		}
 	}
 
-	// 5. Commit grandparent if possible.
-	if err = bc.runCommitRuleOnBestChain(); err != nil {
-		return false, false, nil, errors.Wrap(err, "processBlockPoS: error committing grandparents: ")
+	// Now that we've processed this block, we check for any orphans that
+	// are children of this block and try to process them.
+	orphansAtNextHeight := bc.blockIndexByHeight[blockNode.Height+1]
+	for _, orphan := range orphansAtNextHeight {
+		if orphan.Header.PrevBlockHash.IsEqual(blockNode.Hash) && orphan.IsStored() && !orphan.IsValidated() {
+			// TODO: how do we want to handle failures when validating orphans.
```

Comment:
Seems fine to log the failure and move on. Processing orphans at this point is a best effort thing.